Library/Homebrew/cmd/upgrade.rb

```diff
@@ -41,7 +42,10 @@ def upgrade
       puts pinned.map { |f| "#{f.full_name} #{f.pkg_version}" } * ", "
     end
 
-    outdated.each { |f| upgrade_formula(f) }
+    outdated.each do |f|
+      upgrade_formula(f)
```

Comment:
I just found out that you don't have to tweak `upgrade_formula`. Instead you can tweak the next line to:
```ruby
cleanup_formula(f) if ARGV.include?("--cleanup") && f.installed?
```